gensim/models/word2vec.py

```diff
@@ -909,6 +913,879 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar):
         return sections
 
 
+    def argTopNSorted_nda(self, inArray_nda, topn=1):
+        """
+        Returns an array containing the indexes to produce the topn largest
+        items in sorted order. Runs in n*log(k) time where n is inArray_nda
+        size and k is topn.
+
+        inArray_nda must contain numbers.
+        """
+        if topn == 1:
+            return argmax(inArray_nda)
+        unsortedTopNIdxs_nda = argpartition(-inArray_nda, topn)[:topn]
+        subIdxs_nda = argsort(-inArray_nda[unsortedTopNIdxs_nda])
+        sortedTopNIdxs_nda = unsortedTopNIdxs_nda[subIdxs_nda]
+        return sortedTopNIdxs_nda
+
+
+    def argTopNSorted_2D_nda(self, inArray_nda, topn=1, axis=0):
```

Comment:
yeah, that's one area where I deviate from PEP 8 - I also like to label my variables with their type for convenience - but I'll clean these things up to conform with the package before I'm done.  And I'll also have a look at the built-in gensim.matutils.argsort to see if my version is duplicative (sounds like it may be).  

Do you like the idea of modifying the "most_similar" function to also accept a list of lists for bulk processing?  

My initial effort achieved considerable speed-ups and was designed to be "minimally invasive," but Sebastian showed me a more significant rewrite he had been working on with significantly greater speed-ups, which I agree are well worth the added complexity - so we've merged our approaches.  Do you approve of the path we're on generally?